client/components/tinymce/plugins/wpcom-paste-track/plugin.js

```diff
@@ -0,0 +1,84 @@
+/**
+ * External dependencies
+ */
+import debugFactory from 'debug';
+import tinymce from 'tinymce/tinymce';
+import { recordTracksEvent } from 'state/analytics/actions';
+
+const debug = debugFactory( 'calypso:tinymce-plugins:wpcom-paste-track' );
+
+const SOURCE_GOOGLE_DOCS = 'google_docs';
+const SOURCE_UNKNOWN = 'unknown';
+
+function pasteTrack( editor ) {
+	debug( 'init' );
+
+	const store = editor.getParam( 'redux_store' );
+	if ( ! store ) {
+		debug( 'store not found, do not bind events' );
+		return;
+	}
+
+	/**
+	 * Records the event.
+	 *
+	 * @param {String} types - A string containing the available types for the pasted content, separated by comma.
+	 * @param {String} mode - One of tinyMCE modes: 'html' or 'tinymce'.
+	 */
+	function recordEvent( types, mode ) {
+		debug( 'track paste event ' );
+		store.dispatch( recordTracksEvent( 'calypso_editor_content_paste', {
+			types: types,
+			mode: mode,
```

Comment:
can go ahead and use the object property shorthand here

```js
{ types, mode, source: getSource( … ) }
```

(please ignore other formatting changes)